src/qtism/data/TestPart.php

```diff
@@ -348,20 +348,25 @@ public function getAssessmentSections(): SectionPartCollection
      */
     public function setAssessmentSections(SectionPartCollection $assessmentSections): void
     {
-        if (count($assessmentSections) > 0) {
-            // Check that we have only AssessmentSection and/ord AssessmentSectionRef objects.
-            foreach ($assessmentSections as $assessmentSection) {
-                if (!$assessmentSection instanceof AssessmentSection && !$assessmentSection instanceof AssessmentSectionRef) {
-                    $msg = 'A TestPart contain only contain AssessmentSection or AssessmentSectionRef objects.';
-                    throw new InvalidArgumentException($msg);
-                }
-            }
+        if ($assessmentSections->isEmpty()) {
+            throw new InvalidArgumentException('A TestPart must contain at least one AssessmentSection.');
+        }
 
-            $this->assessmentSections = $assessmentSections;
-        } else {
-            $msg = 'A TestPart must contain at least one AssessmentSection.';
-            throw new InvalidArgumentException($msg);
+        // Check that we have only AssessmentSection and/ord AssessmentSectionRef objects.
```

Comment:
I think this is quite explicit already, maybe we can remove this comment

```suggestion
```

Reply:
*I know it was there before, but it literally reads out laugh the code

Reply:
Done: https://github.com/oat-sa/qti-sdk/pull/367/commits/68bd42b8f627b03eb2e7ad674c99e82ef16d6a30